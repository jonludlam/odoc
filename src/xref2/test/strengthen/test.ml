open Odoc_xref2
open Odoc_xref_test


(* Simple strengthening *)
let name = "Simple strengthening"
let description = {|
This tests that strengthening works. When we strengthen a signature we recursively add
type equations for all abstract types.
|}

let input = {|

type t
type u = t

|}


let simple_strengthening () =
    let p = Common.root_identifier in
    let _, _, sg = Common.model_of_string input in
    let c = Component.Of_Lang.signature [] sg in
    let c' = Strengthen.signature p c in
    let open Format in
    fprintf std_formatter "%s\n%s\n\n" name description;
    fprintf std_formatter "BEFORE\n======\n%!";
    fprintf std_formatter "%a\n%!" Component.Fmt.signature c;
    fprintf std_formatter "AFTER \n======\n%!";
    fprintf std_formatter "%a\n%!" Component.Fmt.signature c'

let _ =
    simple_strengthening ()