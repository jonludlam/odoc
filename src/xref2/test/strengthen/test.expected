Simple strengthening

This tests that strengthening works. When we strengthen a signature we recursively add
type equations for all abstract types.


BEFORE
======
type t/0
type u/1 = t/0

AFTER 
======
type t/2 = [(Root).t]
type u/3 = t/2

